floodlight/models/kinematics.py

```diff
@@ -0,0 +1,329 @@
+import numpy as np
+
+from floodlight import XY
+from floodlight.core.property import PlayerProperty
+from floodlight.models.base import BaseModel
+
+
+class DistanceModel(BaseModel):
+    """Computations for euclidean distances of all players.
+
+    Upon calling the :func:`~DistanceModel.fit`-method, this model calculates the
+    framewise euclidean distance for each player. The following calculations can
+    subsequently be queried by calling the corresponding methods:
+
+        - Frame-wise Distance Covered --> :func:`~DistanceModel.distance_covered`
+        - Cumulative Distance Covered -->
+        :func:`~DistanceModel.cumulative_distance_covered`
+
+    Notes
+    -----
+    The model calculates the distance in the unit of the corresponding position data.
+    This may lead to distortions when used in combination with models that use SI units
+    or when the unit is percent.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from floodlight import XY
+    >>> from floodlight.models.kinematics import DistanceModel
+
+    >>> xy = XY(np.array(((0, 0), (0, 1), (1, 1), (2, 2))))
+
+    >>> dm = DistanceModel()
+    >>> dm.fit(xy)
+
+    >>> dm.distance_covered()
+    PlayerProperty(property=array([[1.        ],
+       [0.70710678],
+       [1.11803399],
+       [1.41421356]]), name='distance_covered', framerate=20)
+    >>> dm.cumulative_distance_covered()
+    PlayerProperty(property=array([[1.        ],
+       [0.70710678],
+       [1.11803399],
+       [1.41421356]]), name='distance_covered', framerate=20)
```

Comment:
where does the framerate come from? xy is instantiated without explicit declaration of a framerate

Reply:
Artefact from prior version, removed